attestation.go

```diff
@@ -38,9 +38,13 @@ func (a *Attestation) Compare(other *Attestation) int {
 		return -1
 	case other == nil:
 		return 1
-	case a.IssuingTime.After(other.IssuingTime):
+	case a.SlotCommitmentID.Index() > other.SlotCommitmentID.Index():
+		return 1
+	case a.SlotCommitmentID.Index() < other.SlotCommitmentID.Index():
+		return -1
+	case a.IssuingTime.Before(other.IssuingTime):
```

Comment:
In this condition you return 1 if `a` is before than `other` on issuingTime, but above you return -1 if `a`'s `SlotCommitmentID.Index()` is before `other`'s commitment index.
The role of this method is for you to evaluate if you should refresh an attestation or not.
As far as I see these checks should be exactly reversed: if `other` commits to an higher index than `a` you return 1, but to avoid refreshing too often I would add additional checks:
1. If the `other`'s block Index is greater than you return 1: you refresh the attestation because you have a block inside a newer slot than the one you previously had.
2. If the IssuingTime of other is greater you return -1. This is exactly to avoid to refresh attestations too often, and always take the earliest block contained in each epoch that also commit to the same slot.